src/data/nat/prime/min_fac.lean

```diff
@@ -0,0 +1,35 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import data.nat.prime
+import tactic.fin_cases
+
+/-!
+# Lemmas about min_fac
+
+These lemmas don't work in `data/nat/prime.lean`, because they use `fin_cases` in the proofs,
+which would introduce a cyclic import.
+-/
+
+namespace nat
+
+lemma min_fac_le_div {n : ℕ} (pos : 0 < n) (np : ¬ prime n) : min_fac n ≤ n / min_fac n :=
+begin
+  by_cases h : n = 1,
+  { subst h, simp, },
+  { replace h : 2 ≤ n := le_of_not_gt (λ hgt, h $ le_antisymm (nat.le_pred_of_lt hgt) (nat.succ_le_of_lt pos)),
```

Comment:
I felt I should be able to do much better than this proof, but it turned out to be a bit frustrating. Still, here is what I came up with using `two_le_iff` (and not `fin_cases`):
```lean
  { rw not_prime_iff_min_fac_lt (two_le_iff.mpr ⟨ne_of_gt pos, h⟩) at np,
    refine min_fac_le_of_dvd _ (div_dvd_of_dvd (min_fac_dvd n)),
    rw two_le_iff,
    have d := min_fac_dvd n,
    split; intro hd,
    { exact ne_of_gt pos (eq_zero_of_div_eq_zero hd d) },
    { exact ne_of_lt np (eq_of_dvd_quot_one d hd) } }
```